config/passport/passport.github.config.js

```diff
@@ -0,0 +1,75 @@
+// Package imports
+const GitHubStrategy = require('passport-github2').Strategy;
+const env = require('dotenv').config();
+const to = require('await-to-js').default;
+
+// Constant imports
+const globalConstants = require('../../constants/constants');
+const configConstants = require('../constants');
+
+// Model imports
+const User = require('../../modules/auth/auth.model');
+
+// Logger import
+const logger = require('../logger/logger');
+
+const passportGithubConfig = (passport) => {
+  passport.serializeUser((user, cb) => {
+    cb(null, user.id);
+  });
+  passport.deserializeUser((id, cb) => {
+    User.findOne({_id: id}).then((user) => cb(null, user));
+  });
+
+  // Creating Github Strategy
+  passport.use(
+      new GitHubStrategy(
+          {
+            clientID: process.env.GITHUB_CLIENT_ID,
+            clientSecret: process.env.GITHUB_CLIENT_SECRET,
+            callbackURL: process.env.GITHUB_CALLBACK_URL,
+          },
+          async (accessToken, refreshToken, profile, done) => {
+            if (!profile.id) {
+              logger.error(configConstants.labels.githubPassportStrategy.githubProfile.failure, profile);
+              return done(null, false, {message: globalConstants.responseMessages.logInUser.socialLogin.networkError});
+            }
+            const [userError, existingUser] = await to(User.findOne({githubId: profile.id}));
+
+            if (userError) {
+              logger.error(configConstants.labels.githubPassportStrategy.findingUser.error, userError);
+              return done(null, false, {message: globalConstants.responseMessages.logInUser.failure});
+            }
+
+            if (existingUser) {
+              logger.info(configConstants.labels.githubPassportStrategy.findingUser.found, existingUser);
+              return done(null, existingUser, {message: globalConstants.responseMessages.logInUser.success});
+            };
+
+            const newUser = new User({
+              githubId: profile?.id,
+              displayName: profile?.displayName,
+              profilePicture: profile?.photos[0]?.value,
+              email: profile?.id + '@email.com',
```

Comment:
```suggestion
              '@email.com'
```
Is this intentional?

Reply:
Yes! Actually, I had this idea even if the user is signing in through google or github, he can change his profile data, add email, set password and can then use the account as the local one. So, when the user sign in through google/github, what he would see as email in its profile data initially?? I cannot set a common email to all because in schema i ahve set the email to be unique. that's why, I am doing this, saving the profile in the email format in place of actual email. then the user can update the email if he wants to.

Reply:
If someone is logging in from Google he/she can't change password/email locally. Please change this implementation and use the information Google is returning. If a user changes the Google profile name this change should be reflected here.
This implementation is not correct if we are authentication users via Google and Github.

Reply:
Done